apps/app/locals.tf

```
@@ -42,18 +42,18 @@ locals {
 
   # Map deployment branch and github environment names for main & testing slot (if enabled)
   branches_and_envs = var.app_config.add_testing_slot ? {
-    "${local.core_branch_name}"    = local.core_gh_env
-    "${local.testing_branch_name}" = local.testing_gh_env
-  } : { "${local.core_branch_name}" = local.core_gh_env }
+    local.core_branch_name    = local.core_gh_env
```

Does this work? I seem to remember it needing something like
```suggestion
    string(local.core_branch_name)    = local.core_gh_env
```
in pretty much any map

---

Works just fine, tested a deployment before raising this :) also validate would catch formatting things now hopefully!

---

Right you are as per the latest run :/ disappointed that validate doesn't catch this

---

Interestingly tflint invoked by terragrunt apply does when ran in the module folder - I shall investigate why pre-commit doesn't pick this up

```
vscode ➜ /workspaces/FlowEHR/apps (jjgriff93/terragrunt-validate-pre-commit) $ terragrunt apply
INFO[0001] Executing hook: before_hook                  
Plugin `terraform` is already installed
Plugin `azurerm` is already installed
Failed to check ruleset; Failed to check `azurerm_container_registry_webhook_invalid_name` rule: app/locals.tf:47,9-31: Ambiguous attribute key; If this expression is intended to be a reference, wrap it in parentheses. If it's instead intended as a literal name containing periods, wrap it in quotes to create a string literal., and 14 other diagnostic(s)
ERRO[0001] Error running hook before_hook with message: Error while running tflint with args: [tflint --init --config /workspaces/FlowEHR/.tflint.hcl /workspaces/FlowEHR/apps] 
ERRO[0001] Errors encountered running before_hooks. Not running 'terraform'. 
ERRO[0001] 1 error occurred:
        * Error while running tflint with args: [tflint --init --config /workspaces/FlowEHR/.tflint.hcl /workspaces/FlowEHR/apps]
 
ERRO[0001] Unable to determine underlying exit code, so Terragrunt will exit with error code 1 
```